pkg/kademlia/routing.go

```diff
@@ -254,8 +259,26 @@ func (rt *RoutingTable) GetBucketTimestamp(bIDBytes []byte) (time.Time, error) {
 	return time.Unix(0, timestamp).UTC(), nil
 }
 
-func (rt *RoutingTable) iterate(opts storage.IterateOptions, f func(it storage.Iterator) error) error {
-	return rt.nodeBucketDB.Iterate(opts, f)
+func (rt *RoutingTable) iterate(start storj.NodeID, f func(storj.NodeID, []byte) error) error {
+	return rt.nodeBucketDB.Iterate(storage.IterateOptions{First: storage.Key(start.Bytes()), Recurse: true},
+		func(it storage.Iterator) error {
+			var item storage.ListItem
+			for it.Next(&item) {
+				nodeID, err := storj.NodeIDFromBytes(item.Key)
+				if err != nil {
+					return err
+				}
+				if nodeID == rt.self.Id {
+					continue //todo:  revisit if self ID should be in routing table
```

Comment:
i think we may want to remove self from the results (see https://github.com/storj/storj/pull/1270/files#diff-06f20e1916ed6b711730b136a3f94975R163 )

Reply:
Its good to see other evidence corroborating this!